corehq/apps/reports/generic.py

```diff
@@ -191,26 +192,29 @@ def __str__(self):
             fields="\n   Report Fields: \n     -%s" % "\n     -".join(self.fields) if self.fields else ""
         )
 
-    def __getstate__(self):
+    def get_json_report_parameters(self):
         """
-            For pickling the report when passing it to Celery.
+            Returns a JSON serializable dict of report parameters for rebuilding the report in Celery.
         """
         request = request_as_dict(self.request)
-
-        return dict(
+        report_state = dict(
```

Comment:
yep it would probably give a JSON error if there was a key with multiple values but after going through all the possible reports we have in cchq, I don't think there's a scenario that would come up anywhere.